### PR TITLE
Differentiate between MXML (Adobe/Apache Flex) files and "normal" XML files

### LINKED ISSUE #82: MXML files cause false positive with default XML rules
The default rules for XML files create a false positive with MXML files (eg: Adobe Flex apps).

MXML files use the ```<![CDATA[ ... ]>``` tags to include code, and this code should be highlighted normally. The syntax coloring for XML files does not color these files well, so this false positive gets annoying because you have to override the entire ```default_syntaxes``` setting or manually set the syntax highlighting every time you open or save a file.

This works for me, but I don't think this rule tweak handles ```.tmLanguage``` files:

    "default_syntaxes": [
        {
            // I put XML first because of files like *.tmLanguage. It is unlikely
            // that this rule will result in a false positive, meaning if it
            // matches, you probably want the XML syntax
            "name": "XML/XML",
            "extensions": ["xml.dist"],
            "rules": [
                {"file_name": ".*\\.xml(\\.dist)?$"}  // <--  Only recognize files with .xml and .xml.dist extensions
            ]
        },
        ...
    ]

I know that the number of users affected by this is small (I may very well be the only one) and Flex is (thankfully) on its way out, but tweaking this rule would be very much appreciated!

Thanks for all of your hard work! :smile:

### PR DESCRIPTION
**Pull request by blag**
Tested in user settings but could use more testing in default settings.

Fixes #82.

Comments welcome.

### CODE REVIEW

File: ApplySyntax.sublime-settings [hunk 1]
```diff
@@ -24,7 +24,18 @@
...
         {
-            // I put XML first because of files like *.tmLanguage. It is unlikely
+            // The MXML rules come before the generic XML rules because they
+            // need syntax highlighting for the CDATA sections, but they do not
+            // have any characteristics that distinguish themselves from XML
+            // aside from their file extension
+            "syntax": "MXML",
+            "extensions": ["mxml"],
+            "rules": [
+                {"first_line": "^<\\?xml"}
```
Comment (facelessuser) [inline ApplySyntax.sublime-settings]: Probably can remove the rules and just use the extensions.  That way xml syntax will be used for files that don't end in mxml.

**Comment by facelessuser** [comment]
> Tested in user settings but could use more testing in default settings.

If you are uncertain, please text more and let me know when you are confident in the change.

**Comment by facelessuser** [comment]
Please make sure you actively communicate in this thread.  If you aren't going to actively participate, in a week, I will mark this as stale.  In another week, I will just close it.  If you are too busy to look at this right now, just let me know when you can get to it.  I just want to make sure this isn't going to be another abandoned pull request where it is assumed that I'll clean it up.

**Comment by blag** [comment]
Sorry, I've been busy and forgot about this. Thank you for the reminder.

I implemented the change you suggested and tested it in my user settings with both `.mxml` and `.xml` files (for false positives). Both worked fine.

I'm happy with the amount of testing I have done.

**Pull request merged**